sirepo_bluesky/sirepo_ophyd.py

```diff
@@ -260,6 +284,35 @@ def create_classes(sirepo_data, connection, create_objects=True):
             base_classes,
             components,
         )
+
+        if el["type"] not in ["mirror", "crystal"]:
+            for k, v in el.items():
+                # TODO: Check mirror and crystal grazing angles
+                if k == "grazingAngle":
+
+                    class tmp(cls):
+                        def set(self, value):
+                            self.grazingAngle.put(value)
+                            ret = self.connection.compute_grazing_orientation(
+                                self.grazingAngle._sirepo_dict
+                            )
+                            # State is added to the ret dict from compute_grazing_orientation and we want to
+                            # make sure the vectors are updated properly every time the grazing angle is updated.
+                            ret.pop("state")
+                            # Update vector components
+                            for cpt in [
+                                "normalVectorX",
+                                "normalVectorY",
+                                "normalVectorZ",
+                                "tangentialVectorX",
+                                "tangentialVectorY",
+                            ]:
+                                getattr(self, cpt).put(ret[cpt])
+                            return NullStatus()
```

Comment:
Maybe we could do this operation on the original class without subclassing, such as suggested in https://stackoverflow.com/a/65571141. @AbbyGi, what do you think?

Reply:
Addressed via https://github.com/NSLS-II/sirepo-bluesky/pull/46/commits/292ed50a9a98878d67942c51443cd83b69dab084.